### PR TITLE
Fix files with Opus audio not being playable in Firefox

### PR DESCRIPTION
**Pull request by szymonrw**
TLDR; Fix playback in Firefox and make `mp4-muxer` more spec-compliant by using `Opus` instead of `opus` for Opus metadata entry.

MP4 files muxed with `mp4-muxer` containing an Opus audio track were playing OK in Safari and Chrome and few other places but not in Firefox.

Firefox uses [mp4parse](https://github.com/mozilla/mp4parse-rust) library to parse MP4 files and it seems to be particularly picky about the Opus sample entry box, only supporting the capitalised `Opus` but not the lower-case `opus` used so far by `mp4-muxer`:

https://github.com/mozilla/mp4parse-rust/blob/1e6e68f329b9d3698075f83b734612877bbecf4d/mp4parse/src/boxes.rs#L168

Playing a file with `opus` in Firefox shows this message in the dev console:

```
Media resource file:///var/home/s/test.mp4 could not be decoded, error: Error Code: NS_ERROR_DOM_MEDIA_METADATA_ERR (0x806e0006)
Details: virtual RefPtr<MP4Demuxer::InitPromise> mozilla::MP4Demuxer::Init(): Parse MP4 metadata failed
```

Using [mpn](https://github.com/sitkevij/mpn) (that uses `mp4parse` library) to parse the file crashes on:

```
read_mp4 failed: InvalidData(StsdBadAudioSampleEntry)
```

Apart from fixing Firefox's issue, this makes `mp4-muxer` more compliant with the spec:

https://web.archive.org/web/20220407094428/https://vfrmaniac.fushizen.eu/contents/opus_in_isobmff.html

See sections [4.2](https://web.archive.org/web/20220407094428/https://vfrmaniac.fushizen.eu/contents/opus_in_isobmff.html#4.2), [4.3.1](https://web.archive.org/web/20220407094428/https://vfrmaniac.fushizen.eu/contents/opus_in_isobmff.html#4.3.1) and [4.6.1](https://web.archive.org/web/20220407094428/https://vfrmaniac.fushizen.eu/contents/opus_in_isobmff.html#4.6.1) all defining it as `Opus`. Tried to look for it mentioning the lower case variant being allowed as well but it seems like that's not technically spec despite some (most?) players handling both as equivalent.

Test examples: (5s of silence)

`opus`: (fails in Firefox)

https://github.com/Vanilagy/mp4-muxer/assets/1305140/b447bd90-94d3-454b-8b14-d13534ea55f2

`Opus`: (works in Firefox)

https://github.com/Vanilagy/mp4-muxer/assets/1305140/ade9ca69-875f-495c-8215-be7c3d69e976

### CODE REVIEW

**Review by Vanilagy — APPROVED** [review]
Thanks for your investigation and fix! Media files are full of these niche details, so I'm happy you fixed this :)

**Comment by Vanilagy** [comment]
Fixed in v3.0.5.

**Comment by szymonrw** [comment]
Thank you for getting it released so quickly!
